### DISCUSSION TITLE
GLFT not loading on android

### DISCUSSION THREAD

**Discussion by rossmarinaro** [enable3d/enable3d#267] [Help / Questions] [answered]
I'm using cordova to build my app and having an issue loading gltfs. I know this works, because it worked before. Now it does not and I have no idea why. I think it may have to do with fetch and local requests, but strangely can't find any solid solution online anywhere. The fact this worked before and now doesn't is confusing and frustrating. If anyone knows why this is happening please let me know.

**Comment by yandeu** [comment]
I just tried it with cordova and capacitor.
With both, it does work on my device but not in the emulator.

**Comment by rossmarinaro** [comment, marked as answer]
add to index.html
`<meta 
      http-equiv="Content-Security-Policy" 
      content="
        default-src * 'self' 'unsafe-inline' 'unsafe-eval' file: data: blob: gap: https://ssl.gstatic.com; 
        connect-src * 'self' file: data: blob: https:; 
        script-src * 'self' file: data: blob: https: 'unsafe-inline' 'unsafe-eval';
        worker-src * 'self' file: data: blob: https:; 
        style-src 'self' 'unsafe-inline'; 
        media-src * 'self' file: data: blob: https:; 
        img-src * 'self' file: data: blob: http:;
        object-src * 'self' file: data: blob: https:;
      "
    >`

**Reply by yandeu** [reply]
Does this also work in the simulator?

**Reply by rossmarinaro** [reply]
I haven't tried with the emulator because it takes too long to load (using the Phaser plugin). on a JSON file I load like so:

models.json:
{
    "gltf": [
        { "cool_asset": "assets/meshes/cool_asset.glb" }
    ]
}

//-----------------------------

in a boot class:

 import resources_3d from './resources/models.json';
 this.load.json('resources_3d', resources_3d);


//----------------------------

and then parse in a preload class:

async parseResources(scene, json)
{
    if (json.hasOwnProperty('gltf')) 
        {
            for (let key in json.gltf) 
            {
                let keys = Object.keys(json.gltf[key]).find(i => i);  
                for (let value in json.gltf[key]) 
                {
                    let values = json.gltf[key][value];  
                    scene.third.load.preload(keys, values); 
                }
            }
        }
}

parseResources(this, this.cache.json.get('resources_3d'));

//--------------------------------------

now you can preload in the main scene like so:
 this.third.load.gltf('cool_asset').then(gltf => /* do stuff */);

I'm using webpack to bundle my app
